diff --git a/node_modules/@cloudflare/next-on-pages/dist/index.js b/node_modules/@cloudflare/next-on-pages/dist/index.js
--- a/node_modules/@cloudflare/next-on-pages/dist/index.js
+++ b/node_modules/@cloudflare/next-on-pages/dist/index.js
@@ -12867,7 +12867,7 @@ async function tryToFixAppRouterNotFoundFunction({
     const invalidError = invalidFunctions.get(
       fullPath.replace(notFoundFuncName, errorFuncName)
     );
-    if (invalidNotFound && !invalidError) {
+    if (invalidNotFound) {
       ignoredFunctions.set(fullPath, {
         reason: "invalid unnecessary not-found function",
         ...fnInfo
